src/AirtableClient.php

```diff
@@ -111,17 +111,47 @@ public function findTheLatest(string $table, $field, ?string $dataClass = null):
         return AirtableRecord::createFromRecord($recordData);
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function addOneRecord(string $table, array $fields, ?string $dataClass = null): ?AirtableRecord
+    {
+        $url = sprintf(
+            '%s/%s',
+            $this->id,
+            $table
+        );
+        $response = $this->request($url, 'POST', $fields);
+
+        $recordData = $response->toArray() ?? null;
+
+        if (!$recordData['id']) {
+            return null;
+        }
+
+        if (null !== $dataClass) {
```

Comment:
duplicate stuff with line 72 maybe move logic in a dedicated method `createRecordFromResponse`

Reply:
I creates a dedicated method createRecordFromResponse but could I have moved some more logic inside?

Reply:
Yes IMO the logic to check `null !== $dataClass` should be inside the new method `createRecordFromResponse`

Reply:
Done !
Thanks bro.